pkg/appgw/certificates_test.go

```diff
@@ -0,0 +1,120 @@
+package appgw
+
+import (
+	"testing"
+
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/annotations"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/tools/cache"
+)
+
+func TestMakeHostToSecretMap(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test setting up SSL redirect annotations")
+}
+
+var _ = Describe("makeHostToSecretMap", func() {
+	Context("looking at TLS specs", func() {
+		cb := makeConfigBuilderTestFixture(nil)
+
+		// Cache a secret
+		cacheKey := getResourceKey(testFixturesNamespace, testFixturesNameOfSecret)
+		const cacheValue = "xyz"
+		cc := cache.NewThreadSafeStore(cache.Indexers{}, cache.Indices{})
+		cc.Add(cacheKey, []byte(cacheValue))
+		cb.k8sContext.CertificateSecretStore = &k8scontext.SecretsStore{
+			Cache: cc,
+		}
+		_ = cb.k8sContext.Caches.Secret.Add(cacheKey)
+
+		ingress := makeIngressFixture()
+
+		// !! Action !!
+		actual := cb.makeHostToSecretMap(ingress)
+
+		expected := map[string]secretIdentifier{
+			"ftp.contoso.com": {
+				testFixturesNamespace,
+				testFixturesNameOfSecret,
+			},
+			"www.contoso.com": {
+				testFixturesNamespace,
+				testFixturesNameOfSecret,
+			},
+			testFixturesHost: {
+				testFixturesNamespace,
+				testFixturesNameOfSecret,
+			},
+			"": {
+				testFixturesNamespace,
+				testFixturesNameOfSecret,
+			},
+		}
+		It("should succeed", func() {
+			Expect(actual).To(Equal(expected))
+		})
+	})
+
+	Context("running", func() {
+		ingress := makeIngressFixture()
+		cb := makeConfigBuilderTestFixture(nil)
+
+		var secrets map[string]secretIdentifier
+
+		// !! Action !!
+		secrets = cb.makeHostToSecretMap(ingress)
+
+		It("should have exact number of keys", func() {
+			Expect(len(secrets)).To(Equal(4))
+		})
+
+		It("should have correct keys", func() {
+			var keys []string
+			for k := range secrets {
+				keys = append(keys, k)
+			}
+			Expect(keys).To(ContainElement(testFixturesHost))
+			Expect(keys).To(ContainElement("www.contoso.com"))
+			Expect(keys).To(ContainElement("ftp.contoso.com"))
+			Expect(keys).To(ContainElement(""))
+		})
+
+		It("has the correct secrets", func() {
+			expected := secretIdentifier{
+				Namespace: testFixturesNamespace,
+				Name:      testFixturesNameOfSecret,
+			}
+			Expect(secrets[testFixturesHost]).To(Equal(expected))
+		})
+	})
+
+	Context("extract host to secret map", func() {
```

Comment:
Maybe need a bit more description here? Trying to understand the difference between this test and the test above "looking at TLS spec"?